File: 3732-calculate-score-after-performing-instructions/3732-calculate-score-after-performing-instructions.cpp
class Solution {
public:
    long long calculateScore(vector<string>& instructions, vector<int>& values) {
        int n = values.size();
        vector<bool> vis(n, false);

        long long sum = 0;
        for (int i = 0; i < n and i >= 0; ) {
            if (vis[i])
                break;
            
            vis[i] = true;
            
            if (instructions[i] == "add") {
                    sum += values[i];
                    i++;
                }
            else 
                i += values[i];
        }
        return sum;
    }
};

File: 3732-calculate-score-after-performing-instructions/3732-calculate-score-after-performing-instructions.cs
public class Solution {
    public long CalculateScore(string[] instructions, int[] values) {
        var n = values.Length;
        long sum = 0;
        var vis = new bool[n];
        for (int i = 0; i >= 0 && i < n && !vis[i]; ) {
            vis[i] = true;
            if (instructions[i].Equals("add")) {
                sum += values[i];
                ++i;
            } else
                i += values[i];
        }
        return sum;
    }
}